app/graphql/update_user_recommendation_preferences_mutation.py

```diff
@@ -12,8 +16,12 @@ class Arguments:
     preferred_topics = graphene.List(Topic, description="Topics that the user expressed interest in")
 
     async def mutate(root, info, input):
-        topics = await TopicModel.get_all()
-        input_topic_ids = {t.id for t in input.preferredTopics}
-        output_topics = [t for t in topics if t.id in input_topic_ids]
+        model = UserRecommendationPreferencesModel(
+            user_id=info.context.get('user_id'),
+            updated_at=datetime.datetime.utcnow(),
+            preferred_topics=await TopicModel.get_topics({t.id for t in input.preferredTopics})
+        )
+
+        await UserRecommendationPreferencesProvider().put(model)
```

Comment:
I like the use of the data mapper pattern. It may be worth leaning into some of the patterns you are using here and giving them names, or otherwise document architecture, such that its clear the strategies that you are using. We could add an `Entity` or `Model` suffix to models which would make it more clear which systems are talking to each other and that we are always passing around domain objects.

(not necessary to be addressed in this PR)